chaoskube/chaoskube.go

```diff
@@ -188,6 +201,23 @@ func (c *Chaoskube) Victim() (v1.Pod, error) {
 	return pods[index], nil
 }
 
+// Victims returns up to N pods as configured by MaxKill flag
+func (c *Chaoskube) Victims() ([]v1.Pod, error) {
+	pods, err := c.Candidates()
+	if err != nil {
+		return []v1.Pod{}, err
+	}
+
+	if len(pods) == 0 {
+		return []v1.Pod{}, errPodNotFound
+	}
+
+	pods = util.PodSubSlice(pods, c.MaxKill)
```

Comment:
We're missing the randomness part from the former `Victim()` function. The `Candidates` list is just a subset of all eligable pods frow which we used to [select one randomly](https://github.com/linki/chaoskube/pull/154/files#diff-634e8348cbd5b2d6a35703a1224c7b51L186).

We should add that functionality to `PodSubSlice` so that it returns a random subset of length `c.MaxKill` from the given pods.

Reply:
There's a test for that kind of stuff [here](https://github.com/linki/chaoskube/blob/master/chaoskube/chaoskube_test.go#L268).

It makes sure that given different seeds different pods get picked. If we don't pick pods randomly the test will fail. It's just a matter of playing with the test seed to make it fail I hope.

Reply:
I removed the previously used `Victim()` function to uncover some tests still using it. I added an `assertVictims()` helper. The randomness test should now correctly fail.

Reply:
The `PodSlice` util function could look something like this:

```golang
func PodSubSlice(pods []v1.Pod, count int) []v1.Pod {
	subset := []v1.Pod{}

	maxCount := len(pods)
	if count > maxCount {
		count = maxCount
	}

	for i := 0; i < count; i++ {
		index := rand.Intn(len(pods))
		subset = append(subset, pods[index])
	}

	return subset
}
```

Although that could pick the same pod twice. Whatever you prefer.

Reply:
Thanks for the nice shuffle solution.